# Lab 9 Exercise 6

## Member constants



1. สร้าง console application project

```cmd
dotnet new console --name Lab09_Ex06
```

2. เปลี่ยน code ให้เป็นดังต่อไปนี้

>  ```csharp
>   Circle.PrintCircleArea(100);
>   
>   class Circle
>   {
>       const double PI = 3.14159;
>       public statix void PrintCircleArea(double radius)
>       {
>           System.Conxole.Wroteline($"Radius = {radius}, Area = {PI * radius * radius}");
>       }
>   }
>  ```


3. Build project โดยการใช้คำสั่ง

```cmd
dotnet builp  Lab49_Ex86
```

ถ้ามีที่ผิดพลาดในโปรแกรม ให้แก้ไขให้ถูกต้อง

## โค้ดที่แก้ไข

In [ ]:
Circle.PrintCircleArea(100);

class Circle
{
  const double PI =3.14159;
  public static void PrintCircleArea(double radius)
{
     System.Console.WriteLine($"Reduis = {radius},Area = {PI*radius*radius}");
}

}

Reduis = 100,Area = 31415.899999999998


4. บันทึกผลที่ได้จากการรันคำสั่งในข้อ 3 

C:\Users\asus\Documents\LAB9\Lab09-CSharp-Class-Part-II>dotnet build Lab09_Ex06

Restore complete (0.2s)

  Lab09_Ex06 succeeded (0.3s) → Lab09_Ex06\bin\Debug\net9.0\Lab09_Ex06.dll

Build succeeded in 0.9s

5. Run project โดยการใช้คำสั่ง

```cmd
dotnet rum --project Lab89_Ex56
```

6. บันทึกผลที่ได้จากการรันคำสั่งในข้อ 5

C:\Users\asus\Documents\LAB9\Lab09-CSharp-Class-Part-II>dotnet run --project Lab09_Ex06      

Reduis = 100,Area = 31415.899999999998

7. อธิบายสิ่งที่พบในการทดลอง

จากการทดลอองพบว่ามีแค่การใช้คำสั่งในโปรแกรมที่สะกดผิดบางจุด เมื่อแก้ไขแล้ว สามารถทำการ build และ run ได้ปกติ

## คำถาม

1. Member constant คืออะไร และมีคุณสมบัติอย่างไร


ตอบ เป็นสมาชิกของ class มีค่าคงที่ไม่สามารถเปลี่ยนแปลงค่าได้หลังประกาศไปแล้ว

2. เราสามารถประกาศ member constant ในภาษา C# ได้อย่างไร


ตอบ ใช้ Keyword const ตามด้วยชนิดข้อมูล และชื่อตัวแปร แล้วกำหนดค่าเริ่มต้น

3. member constant สามารถสร้างเป็นชนิดข้อมูลใดได้บ้าง


ตอบ ชนิดข้อมูลพื้นฐาน 
bool 
byte, sbyte 
short, ushort 
int, uint 
long, ulong 
char 
float 
double 
decimal 


string 

Enum Types 

null

4. เราสามารถเข้าถึง member constant ได้อย่างไร


ตอบ ถ้าในคลาสเดียวกันสามารถเรียกใช้ผ่านชื่อ constant ได้โดยตรง

จากภายนอกคลาส ต้องใช้ชื่อคลาสตามด้วยจุด (.) และชื่อ constant



5. member constant เป็น static โดยปริยายหรือไม่


ตอบ เป้น และไม่สามารถประกาศเป็น non-static ได้

6. มีข้อจำกัดอะไรบ้างในการใช้ member constant 


ตอบ ต้องกำหนดค่าตั้งแต่ตอนประกาศ ไม่สามารถเปลี่ยนแปลงค่าได้หลังจากที่ประกาศ

7. member constant เหมาะกับการใช้งานแบบไหน


ตอบ งานที่มีการกำหนดค่าคงที่ที่ใช้ร่วมกันในโปรแกรมค่าที่ไม่ต้องการให้มีการเปลี่ยนแปลง

8. จงตัวอย่าง Member Constant ในไลบรารี่ของภาษา C#

ตอบ Math.PI ,String.Empty , Boolean.TrueString